udemy_exercises/songs/Album.java

```diff
@@ -1,5 +1,71 @@
-package udemy_exercises.songs;
 
-class Album {
+import java.util.ArrayList;
+import java.util.LinkedList;
+
+public class Album {
+    private String name;
+    private String artist;
+    private ArrayList<Song> songs;
+
+    public Album(String name, String artist) {
+        this.name = name;
+        this.artist = artist;
+        this.songs = new ArrayList<Song>();
+    }
+
+    public boolean addSong(String title, double duration) {
+        if(findSong(title) == null) {
+            this.songs.add(new Song(title, duration));
+            return true;
+        }
+        return false;
+    }
+
+    private Song findSong(String title) {
+        for(Song checkedSong: this.songs) {
+            if(checkedSong.getTitle().equals(title)) {
+                return checkedSong;
+            }
+        }
+        return null;
+    }
+
+    public boolean addToPlayList(int trackNumber, LinkedList<Song> playList) {
+        int index = trackNumber -1;
+        if((index >0) && (index <= this.songs.size())) {
```

Comment:
Consider the ArrayList<Song> in the Album contains three Songs.  The client absentmindedly requests the fourth Song (track #4) of the Album to be added to the LinkedList<Song> provided, so calls this method with trackNumber = 4.  What will be the result?  (there's a defect here)

You have two conditions on line 35:  index is valid (i.e. not a negative number), and the index <= the size of the LinkedList.  in the first case, what your really confirming is that the trackNumber provided didn't result in a negative number.  If subtracting one from the provided trackNumber will result in a negative number, does the code need to subtract one (1) in order to discover this?  Consider a refactoring which would bypass the entire method in the case the trackNumber itself is invalid.